Fixes for tpm2_getmanufec

martinezjavier opened the pull request:
When testing the tool I noticed that it some bugs. These commits fixes them.

williamcroberts commented on tools/tpm2_getmanufec.c:
```diff
@@ -184,36 +184,35 @@ int createEKHandle(TSS2_SYS_CONTEXT *sapi_context)
+        } else {
+            sessionData.hmac.t.size = sizeof(sessionData.hmac) - 2;
+
+            if (tpm2_util_hex_to_byte_structure(ctx.endorse_passwd,
```
Oooh old -X option code...

williamcroberts replied:
Update this to use the password util routines, that's my only request for this PR, the rest looks great.

martinezjavier commented:
@williamcroberts Ok, since is an independent change I've added it as a separate commit. Also while testing the password options, I noticed that the endorsement password wasn't working so I fixed it before doing the conversion.

And also added a test case to ensure that the password options won't regress in the future and cleaned the test a bit.

williamcroberts commented:
Fail Snippet:
```
WARN: TLS communication with the said TPM manufacturer server setup with SSL_NO_VERIFY!
persistent-handle[0]:0x81000000 key-alg:rsa hash-alg:sha256 object-attr:0x300b2
tpm2_evictcontrol -H $handle -A o $opass on line 80 failed: 1
[31m test_tpm2_getmanufec.sh Fail [0m
Success: setting up the number of allowed tries in the lockout parameters
```

It passed on CLANG, so I restarted the build.

williamcroberts commented on tools/tpm2_getmanufec.c:
```diff
@@ -582,13 +526,19 @@ static bool on_option(char key, char *value) {
+            return false;
+        }
+
+        return_val = tpm2_util_string_to_uint32(value, &ctx.owner_session_data.sessionHandle);
```
This could be a lot more compact:
```
UINT32 tmp;
return_val = tpm2_util_string_to_uint32(value, &tmp);
if (error)...
fi
ctx.owner_session_data.sessionHandle =
ctx.endorse_session_data.sessionHandle = tmp;
```
or
```
return_val = tpm2_util_string_to_uint32(value, &ctx.endorse_session_data.sessionHandle)
                    && tpm2_util_string_to_uint32(value, &ctx.owner_session_data.sessionHandle);
if (error)....
fi
```

martinezjavier replied:
indeed, I wasn't thinking clearly.

williamcroberts commented:
@martinezjavier looks broken with GCC

martinezjavier commented:
@williamcroberts yes, not sure why though. I'll investigate.

martinezjavier commented:
@williamcroberts the CI test passes now.

I just pushed a new version since found an error in the updated man page, but should be ready to be merged.
